specs/ChainSpec/LightClient.md

```diff
@@ -0,0 +1,86 @@
+# Light Client
+
+The state of the light client is defined by:
+
+1. `BlockHeaderInnerLiteView` for the current head (which contains `height`, `epoch_id`, `next_epoch_id`, `prev_state_root`, `outcome_root`, `timestamp` and the hash of the block producers set for the next epoch `next_bp_hash`);
+2. The set of block producers for the current and next epochs.
+
+Light clients operate by periodically fetching instances of `LightClientBlockView` via particular RPC end-point desdribed [below](#rpc-end-point).
+
+```rust
+pub struct ValidatorStakeView {
+    pub account_id: AccountId,
+    pub public_key: PublicKey,
+    pub stake: Balance,
+}
+
+pub struct LightClientBlockView {
+    pub prev_block_hash: CryptoHash,
+    pub next_block_inner_hash: CryptoHash,
+    pub inner_lite: BlockHeaderInnerLiteView,
+    pub inner_rest_hash: CryptoHash,
+    pub next_bps: Option<Vec<ValidatorStakeView>>,
+    pub approvals_next: Vec<Option<Signature>>,
+    pub approvals_after_next: Vec<Option<Signature>>,
+}
+```
+
+Recall that the hash of the block is
+
+```rust
+sha256(concat(
+    prev_hash,
+    sha256(concat(
+        sha256(borsh(inner_lite)),
+        sha256(borsh(inner_rest))
+    ))
+))
+```
+
+The fields `prev_block_hash`, `next_block_inner_hash` and `inner_rest_hash` are used to reconstruct the hashes of the current and next block in the following way (where `block_view` is an instance of `LightClientBlockView`):
+
+```rust
+current_block_hash = sha256(concat(
+    block_view.prev_block_hash,
+    sha256(concat(
+        sha256(borsh(block_view.inner_lite)),
+        block_view.inner_rest_hash,
+    ))
+))
+
+next_block_hash = sha256(concat(
+    current_block_hash,
+    block_view.next_block_inner_hash
+))
+```
+
+The light client updates its head with the information from `LightClientBlockView` iff:
+
+1. The height of the block is higher than the height of the current head;
+2. The epoch of the block is equal to the `epoch_id` or `next_epoch_id` known for the current head;
+3. If the epoch of the block is equal to the `next_epoch_id` of the head, then `next_bps` is not `None`;
```

Comment:
If you first receive block from E+1, if all the conditions here are satisfied, it will get accepted, otherwise it will get rejected.
If it got accepted, and you later receive a block from E, that block from E will be rejected, because it will violate (1).